### PR TITLE
#1417 - get many work packages endpoint

### PR DESCRIPTION
**Pull request by LOTaher**
## Changes

Created a new work packages endpoint.

## Notes

Having trouble testing the endpoint. Still not too sure if the code works properly.

## To Do

- [ ] Write tests
- [ ] Test endpoints using Postman

## Checklist

- [x ] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #1417 )

### CODE REVIEW

**Review by ryanhowe28 — COMMENTED** [review]
Here is a video for how to use postman to check if your code is doing what its supposed to. https://www.youtube.com/watch?v=CgFTb_hKdEo
This looks good so far, I would recommend looking at the getManyTeams or Projects endpoints that are similar if you are having any trouble.

**Review by RChandler234 — CHANGES_REQUESTED** [review]

File: src/backend/src/services/work-packages.services.ts [hunk 1]
```diff
@@ -108,6 +108,55 @@ export default class WorkPackagesService {
...
   }
 
+  /**
+   * Retrieve a subset of work packages.
+   * @param wbsNums the WBS numbers of the work packages to retrieve
+   * @returns the work packages with the given WBS numbers
+   * @throws if any of the work packages are not found
+   */
+  static async getManyWorkPackages(wbsNums: WbsNumber[]): Promise<WorkPackage[]> {
+    for (const wbsNum of wbsNums) {
```
Comment (RChandler234) [inline src/backend/src/services/work-packages.services.ts:118]: you can do wbsNums.forEach() here, it's a bit cleaner

File: src/backend/src/services/work-packages.services.ts [hunk 2]
```diff
@@ -108,6 +108,55 @@ export default class WorkPackagesService {
...
+                projectNumber: wbsNum.projectNumber,
+                workPackageNumber: wbsNum.workPackageNumber
+              }
+            }
+          ]
+        },
+        ...workPackageQueryArgs
+      });
+
+      console.log(workPackage);
```
Comment (RChandler234) [inline src/backend/src/services/work-packages.services.ts:148]: remove

File: src/backend/src/routes/work-packages.routes.ts [hunk 3]
```diff
@@ -6,6 +6,15 @@ import { intMinZero, isDate, isWorkPackageStageOrNone, nonEmptyString } from '..
 const workPackagesRouter = express.Router();
 
 workPackagesRouter.get('/', WorkPackagesController.getAllWorkPackages);
+workPackagesRouter.get(
+  '/get-many',
+  body('wbsNums').exists().isArray(),
```
Comment (RChandler234) [inline src/backend/src/routes/work-packages.routes.ts:11]: you shouldn't need .exists here

**Review by RChandler234 — CHANGES_REQUESTED** [review]

**Review by RChandler234 — APPROVED** [review]
LGTM 🔥

**Pull request merged**